### DISCUSSION TITLE
Am I able to use the old Tauri v1 IPC method in Tauri v2?

### DISCUSSION THREAD

**Discussion by SpikeHD** [tauri-apps/tauri#10224] [Q&A] [answered]
As I understand it, the new IPC method uses custom protocols (very cool! big fan). Unfortunately, this is incompatible with a project of mine that displays an external site, and is very strict in it's CSP/CORs in a way I cannot change :( (unless the answer [here](https://github.com/tauri-apps/tauri/discussions/6187) has changed haha)

Someone else worked briefly on a port from v1 -> v2 and this was the largest problem for them, as much of the functionality is facilitated through IPC. When they asked me about it, I dug through the Tauri source a bit to see if there was a configuration I could change, or a flag to set, or something, but I couldn't find anything. This was also a couple months ago so things may have changed since then lol. Does Tauri still have the old functionality in there somewhere as a fallback? Can I force that fallback somehow?

I understand that Tauri is not an "external-site-first" library so I get that this is prooooobably not a huge compatibility consideration, but if there is some way I can use the old method so I can port my project and benefit from all the new stuff, please let me know!

**Comment by FabianLars** [comment, marked as answer]
> Unfortunately, this is incompatible with a project of mine that displays an external site, and is very strict in it's CSP/CORs in a way I cannot change :( (unless the answer https://github.com/tauri-apps/tauri/discussions/6187 has changed haha)

Tauri is supposed to fall back to the old implementation automatically on csp/cors issues. There is no force switch but if you have something reproducible showing problems with the auto fallback we'd appreciate any more info we can get.

**Reply by SpikeHD** [reply]
The fork that the other person was working on - while I very much appreciate their work - is a bit messy, so I will start a new v2 branch for the port and get back to you when I get it to a compiling state :+1:

**Reply by SpikeHD** [reply]
@FabianLars I have gotten it to a compile-able state (though WebView2 will sometimes spit out a STATUS_ACCESS_VIOLATION page because of my custom initialization script. A problem for another day!). When the page loads, the following is logged to the top of the console:

![image](https://github.com/tauri-apps/tauri/assets/25207995/d5ee3321-e57b-4456-8cf5-8132313148cd)

I will note that `__TAURI__.core.invoke()` exists and seems to work, so whatever problem existed before does not exist now :)

That notification thing seems problematic though. You can try this yourself by taking a look at the [Dorion/next](https://github.com/SpikeHD/Dorion/tree/next) branch, or opening DevTools in a [build Artifact](https://github.com/SpikeHD/Dorion/actions/workflows/build.yml?query=branch%3Anext+event%3Apush) if you're curious, though I am sure this is minimally reproducible.

**Reply by FabianLars** [reply]
> That notification thing seems problematic though.

That depends. Does everything work or is something broken by it? The IPC will try to use the v2 impl first and fall back to the v1 one on an error. Edit: That's not 100% true, there are conditions where it skips right to the v1 but i can't remember the details.

If there are issues, it may be because `dangerousRemoteDomainIpcAccess` wasn't migrated, you can do it yourself like this: https://v2.tauri.app/security/capabilities/#remote-api-access
